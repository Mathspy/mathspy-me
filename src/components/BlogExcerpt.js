import React from "react";
import { Link } from "gatsby";

import { css } from "emotion";

import theme from "../theme";

const BlogExcerpt = ({ title, excerpt, to }) => {
  return (
    <div className={`${blogStyles}`}>
      <Link to={to}>
        <h1>{title}</h1>
        <p>{excerpt}</p>
      </Link>
    </div>
  );
};

const blogStyles = css`
  a {
    text-decoration: none;
    color: black;
  }

  &:hover {
    background-color: #f895544d;
  }

  p,
  ul > li {
    line-height: 1.7;
  }

  p > code {
    background-color: ${theme.mainColorLightened};
  }
`;

export default BlogExcerpt;
