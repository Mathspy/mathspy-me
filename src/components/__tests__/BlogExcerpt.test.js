import React from "react";
import { Link } from "gatsby";

import { shallow } from "enzyme";

import BlogExcerpt from "../BlogExcerpt";

describe("BlogExcerpt", () => {
  it("should be a div containing a single GatsbyLink", () => {
    const wrapper = shallow(<BlogExcerpt />);

    expect(wrapper.is("div")).toBe(true);
    expect(wrapper.children().is(Link)).toBe(true);
  });

  it("should display the title passed to it", () => {
    const wrapper = shallow(<BlogExcerpt title="testtitle" />);

    expect(wrapper.find("h1").text()).toBe("testtitle");
  });

  it("should display the excerpt passed to it", () => {
    const wrapper = shallow(<BlogExcerpt excerpt="testexcerpt" />);

    expect(wrapper.find("p").text()).toBe("testexcerpt");
  });

  it("should match snapshot", () => {
    const wrapper = shallow(<BlogExcerpt />);

    expect(wrapper).toMatchSnapshot();
  });
});
