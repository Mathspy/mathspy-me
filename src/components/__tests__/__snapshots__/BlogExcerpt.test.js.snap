// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogExcerpt should match snapshot 1`] = `
.emotion-0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: black;
}

.emotion-0:hover {
  background-color: #f895544d;
}

.emotion-0 p,
.emotion-0 ul > li {
  line-height: 1.7;
}

.emotion-0 p > code {
  background-color: #c697e24d;
}

<div
  className="emotion-0"
>
  <Component>
    <h1 />
    <p />
  </Component>
</div>
`;
